public/app/features/correlations/utils.ts

```diff
@@ -83,6 +84,27 @@ const decorateDataFrameWithInternalDataLinks = (dataFrame: DataFrame, correlatio
   });
 };
 
+/*
+If a correlation was made based on the log line field prior to the loki data plane, they would use the field "Line" 
+
+Change it to use whatever the body field name is post-loki data plane
+*/
+const fixLokiDataplaneFields = (correlations: CorrelationData[], dataFrame: DataFrame) => {
+  return correlations.map((correlation) => {
+    if (
+      correlation.source.meta?.id === 'loki' &&
+      config.featureToggles.lokiLogsDataplane === true &&
+      correlation.config.field === 'Line'
+    ) {
+      const logsFrame = parseLogsFrame(dataFrame);
+      if (logsFrame != null && logsFrame.bodyField.name !== undefined) {
```

Comment:
Would it make sense to skip switching to `body` in case the `Line` field exists? Not sure if it's possible that both would exist at the same time 🤔

Reply:
Oh yes, both should be supported because `lokiLogsDataplane` is not GA https://github.com/grafana/grafana/blob/main/pkg/services/featuremgmt/registry.go#L344

Reply:
Taking my comment back as I misread.